LambdaTimeline/View Controllers/ImagePostViewController.swift

```diff
@@ -114,6 +124,112 @@ class ImagePostViewController: ShiftableViewController {
         view.layoutSubviews()
     }
     
+    @IBAction func blurSliderChanged(_ sender: Any) {
+        // UIImage -> CGImage (Core Graphics) -> CIImage
+        guard let image = originalImage else { return }
+        
+        imageView.image = blurImage(image)
+    }
+    
+    @IBAction func processingTypeChanged(_ sender: Any) {
+        switch processingSegementedControl.selectedSegmentIndex {
+            case 0:
+                blurSlider.isHidden = false
+                if let originalImage = originalImage {
+                    imageView.image = blurImage(originalImage)
+                }
+            case 1:
+                blurSlider.isHidden = true
+                if let originalImage = originalImage {
+                    imageView.image = comicImage(originalImage)
+                }
+            case 2:
+                blurSlider.isHidden = true
+                if let originalImage = originalImage {
+                    imageView.image = tileImage(originalImage)
+                }
+            case 3:
+                blurSlider.isHidden = true
+                if let originalImage = originalImage {
+                    imageView.image = kaleidoscopeImage(originalImage)
+                }
+            default:
+            break
+            
+        }
+    }
+    
+    func comicImage(_ image: UIImage) -> UIImage? {
+        guard let cgImage = image.cgImage else { return nil }
+        let ciImage = CIImage(cgImage: cgImage)
+        
+        // Filter
+        let filter = CIFilter.comicEffect()
+        filter.setValue(ciImage, forKey: kCIInputImageKey)
+        
+        // Render the image
+        guard let outputCIImage = filter.outputImage else { return nil }
+        guard let outputCGImage = context.createCGImage(outputCIImage, from: CGRect(origin: .zero, size: image.size)) else { return nil }
+        
+        // CIImage -> CGImage -> UIImage
+        return UIImage(cgImage: outputCGImage)
+        
+    }
```

Comment:
It would be good to cleanup comments so they match, and are not redundant, or wrong in order (mine were a bit off by the end of class).

Some of the comments from class were for instruction purposes, and not something you need to put into your final code.

```swift
func comicImage(_ image: UIImage) -> UIImage? {
    guard let cgImage = image.cgImage else { return nil }
    let ciImage = CIImage(cgImage: cgImage)
    
    let filter = CIFilter.comicEffect()
    filter.setValue(ciImage, forKey: kCIInputImageKey)
    filter.inputImage = ciImage
    
    guard let outputCIImage = filter.outputImage else { return nil }
    guard let outputCGImage = context.createCGImage(outputCIImage, from: CGRect(origin: .zero, size: image.size)) else { return nil }
    return UIImage(cgImage: outputCGImage)
}
```